execute/report/builder.go

```diff
@@ -66,21 +72,16 @@ type execReportBuilder struct {
 func (b *execReportBuilder) Add(
 	commitReport plugintypes.ExecutePluginCommitDataWithMessages,
 ) (plugintypes.ExecutePluginCommitDataWithMessages, error) {
-	// TODO: buildSingleChainReportMaxSize needs to be part of the builder in order to access state.
-	execReport, encodedSize, updatedReport, err :=
-		buildSingleChainReportMaxSize(b.ctx, b.lggr, b.hasher, b.tokenDataReader, b.encoder,
-			commitReport,
-			int(b.maxReportSizeBytes-b.reportSizeBytes))
+	execReport, updatedReport, err := b.buildSingleChainReport(b.ctx, commitReport)
 
```

Comment:
Notice that the accumulator is gone, it's now baked into `buildSingleChainReport` and stored as part of an accumulator object.